### PR TITLE
GH-2101 set fedx:writable property for explicit writable member endpoint

### PR DESCRIPTION
**Pull request by jeenbroekstra**
GitHub issue resolved: #2101  

Briefly describe the changes proposed in this PR:

Added `fedx:writable` property which can optionally be set to true for a member endpoint service (or a member repository when using "classic config").

Added simple unit tests to verify property is communicated. 

---- 
PR Author Checklist: 

 - [x] my pull request is [self-contained](https://rdf4j.org/documentation/developer/merge-strategy/#self-contained-changes-pull-requests-and-commits)
 - [x] I've added tests for the changes I made
 - [x] every commit message starts with the issue number (GH-xxxx) followed by a meaningful description of the change
 - [x] every commit has been [signed off](https://stackoverflow.com/questions/1962094/what-is-the-sign-off-feature-in-git-for)

Note: we merge all feature pull requests using [squash and merge](https://help.github.com/en/github/administering-a-repository/about-merge-methods-on-github#squashing-your-merge-commits). See [RDF4J git merge strategy](https://rdf4j.org/documentation/developer/merge-strategy/) for more details.

### CODE REVIEW

**Comment by jeenbroekstra** [comment]
> LGTM, nice and clean change to add writable configurability via the repository configuration

Thanks, good to hear. 
 
> The next step is probably to also allow influencing the write strategy to pick a user-configured write target

Do you mean a way to set up a federation with a hand-picked write target programmatically? It's technically already possible of course, but I'll see if I can make it a bit more convenient by extending the FedXFactory (as an aside: that should really be called a builder rather than a factory, any objections if I deprecate/rename?).

**Comment by jeenbroekstra** [comment]
@aschwarte10 I'd like to hear your thoughts on the latest changes. Anything I overlooked in terms of making per-endpoint configuration of writability more easily accessible?

One thing to note is that we don't enforce anywhere that only a single repository can be writable: if a user configuresmore than one, fedx will just pick the first one it encounters (similar to how this already worked). This is deliberate as I don't want to put checks in at config time that we may need to remove again later, when we look at strategies involving multiple target repos.

File: tools/federation/src/main/java/org/eclipse/rdf4j/federated/write/WriteStrategy.java [hunk 1]
```diff
@@ -25,49 +24,35 @@
...
-	public void initialize() throws RepositoryException;
-
-	/**
-	 * Returns true if this instance is initialized
-	 * 
-	 * @return flag indicating the initialization resources
-	 */
-	public boolean isInitialized();
-
+public interface WriteStrategy extends AutoCloseable {
```
Comment (aschwarte10) [inline tools/federation/src/main/java/org/eclipse/rdf4j/federated/write/WriteStrategy.java:27]: we should mention here that an instance should be specifically bound to a RepositoryConnection, at least as internal comment

**Comment by aschwarte10** [comment]
> @aschwarte10 I'd like to hear your thoughts on the latest changes. Anything I overlooked in terms of making per-endpoint configuration of writability more easily accessible?
> 
> One thing to note is that we don't enforce anywhere that only a single repository can be writable: if a user configuresmore than one, fedx will just pick the first one it encounters (similar to how this already worked). This is deliberate as I don't want to put checks in at config time that we may need to remove again later, when we look at strategies involving multiple target repos.

Change looks all good to me. We require a bit more testing in action though

**Comment by aschwarte10** [comment]
> 
> Do you mean a way to set up a federation with a hand-picked write target programmatically? It's technically already possible of course, but I'll see if I can make it a bit more convenient by extending the FedXFactory.

Makes perfectly sense to switch to a builder pattern, might be a good time to do this now when we introduce new configuration settings anyways

**Comment by jeenbroekstra** [comment]
I'll keep the builder pattern idea in reserve for now, priority on merging this change.

**Pull request merged**